source/extensions/quic_listeners/quiche/platform/quic_mem_slice_span_impl.h

```diff
@@ -26,51 +27,82 @@ class QuicMemSliceSpanImpl {
    */
   explicit QuicMemSliceSpanImpl(Envoy::Buffer::Instance& buffer) : buffer_(&buffer) {}
   explicit QuicMemSliceSpanImpl(QuicMemSliceImpl* slice) : buffer_(&slice->single_slice_buffer()) {}
+  explicit QuicMemSliceSpanImpl(absl::Span<QuicMemSliceImpl> slices) : span_(slices) {}
 
   QuicMemSliceSpanImpl(const QuicMemSliceSpanImpl& other) = default;
   QuicMemSliceSpanImpl& operator=(const QuicMemSliceSpanImpl& other) = default;
 
-  QuicMemSliceSpanImpl(QuicMemSliceSpanImpl&& other) noexcept : buffer_(other.buffer_) {
+  QuicMemSliceSpanImpl(QuicMemSliceSpanImpl&& other) noexcept
+      : buffer_(other.buffer_), span_(std::move(other.span_)) {
     other.buffer_ = nullptr;
   }
 
   QuicMemSliceSpanImpl& operator=(QuicMemSliceSpanImpl&& other) noexcept {
     if (this != &other) {
       buffer_ = other.buffer_;
       other.buffer_ = nullptr;
+      span_ = std::move(other.span_);
     }
     return *this;
   }
 
   // QuicMemSliceSpan
   quiche::QuicheStringPiece GetData(size_t index);
-  QuicByteCount total_length() { return buffer_->length(); };
-  size_t NumSlices() { return buffer_->getRawSlices(nullptr, 0); }
+  QuicByteCount total_length() {
+    if (buffer_ != nullptr) {
+      return buffer_->length();
+    } else {
+      size_t len = 0;
```

Comment:
if we are returning a QuicByteCount rather than size_t should we use that for the accumulation?